spec/pane-spec.coffee

```diff
@@ -281,15 +281,17 @@ describe "Pane", ->
         describe "when the item has no uri", ->
           it "presents a save-as dialog, then saves the item with the given uri before removing and destroying it", ->
             itemURI = null
-
             showSaveDialog.andReturn("/selected/path")
             confirm.andReturn(0)
+            spyOn(pane, 'saveItemAs')
             pane.destroyItem(item1)
-
-            expect(showSaveDialog).toHaveBeenCalled()
-            expect(item1.saveAs).toHaveBeenCalledWith("/selected/path")
-            expect(item1 in pane.getItems()).toBe false
-            expect(item1.isDestroyed()).toBe true
+            setTimeout( ->
```

Comment:
At the very least you can do a Jasmine-style `waitsFor "item to be destroyed", -> item1.isDestroy()` to hold the test back until that condition is satisfied. Or do an `onDidDestroy` on the item and a `waitsFor` block that takes a parameter.